dotCMS/src/test/java/com/dotcms/filters/NormalizationFilterTest.java

```diff
@@ -0,0 +1,115 @@
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+public class NormalizationFilterTest extends UnitTestBase {
+
+    private final static NormalizationFilter normalizationFilter = new NormalizationFilter();
+    private static HttpServletResponse response;
+    private static FilterChain chain;
+    private static ArgumentCaptor<HttpServletRequest> capturedRequest;
+
+    @BeforeClass
+    public static void prepare() throws IOException, ServletException {
+
+        //Response
+        response = mock(HttpServletResponse.class);
+
+        //Chain
+        chain = mock(FilterChain.class);
+        capturedRequest = ArgumentCaptor
+                .forClass(HttpServletRequest.class);
+        //Capturing the request when is passed down to the chain
+        doNothing().when(chain).doFilter(capturedRequest.capture(), any(HttpServletResponse.class));
+    }
+
+    /**
+     * Test to verify the {@link NormalizationFilter} is applying properly the normalization on
+     * invalids URIs
+     */
+    @Test
+    public void test_uri_normalization_invalid_URI() throws IOException, ServletException {
+
+        // A ".." segment is removed only if it is preceded by a non-".." segment
+        String originalURI = "/test/../folder/important/secret_file.dat";
+        String expectedNormalizedURI = "/folder/important/secret_file.dat";
+        compare(originalURI, expectedNormalizedURI);
+
+        // A ".." segment is removed only if it is preceded by a non-".." segment
+        originalURI = "/test/../folder/folder1/forward_jsp.jsp?FORWARD_URL=http://google.com";
+        expectedNormalizedURI = "/folder/folder1/forward_jsp.jsp?FORWARD_URL=http://google.com";
+        compare(originalURI, expectedNormalizedURI);
+
+        // A ".." segment is removed only if it is preceded by a non-".." segment
+        originalURI = "/test/../folder/important/../secret_file.dat";
+        expectedNormalizedURI = "/folder/secret_file.dat";
+        compare(originalURI, expectedNormalizedURI);
+
+        // Each "." segment is simply removed
+        originalURI = "./folder/folder1/file.dat";
+        expectedNormalizedURI = "folder/folder1/file.dat";
+        compare(originalURI, expectedNormalizedURI);
+
+        // Each "." segment is simply removed
+        originalURI = "./folder/./folder1/file.dat";
+        expectedNormalizedURI = "folder/folder1/file.dat";
+        compare(originalURI, expectedNormalizedURI);
+
+        // Each "." segment is simply removed
+        originalURI = "/folder/./folder1/file.dat";
+        expectedNormalizedURI = "/folder/folder1/file.dat";
+        compare(originalURI, expectedNormalizedURI);
+    }
+
+    /**
+     * Test to verify the {@link NormalizationFilter} is applying properly the normalization on
+     * valid URIs
+     */
+    @Test
+    public void test_uri_normalization_valid_URI() throws IOException, ServletException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4798850)